crypto/signatures.go

```diff
@@ -31,7 +31,7 @@ func SignatureAlgorithm(k gocrypto.PublicKey) sigpb.DigitallySigned_SignatureAlg
 		return sigpb.DigitallySigned_ECDSA
 	case *rsa.PublicKey:
 		return sigpb.DigitallySigned_RSA
-	case *ed25519.PublicKey:
+	case ed25519.PublicKey:
```

Comment:
Why is this different to the other?

Reply:
The underlying type is `[]byte`. https://golang.org/src/crypto/ed25519/ed25519.go?s=1385:1406#L31

The other types are `struct`s that have pointers. 

This change looks correct